rlclientlib/factory_resolver.cc

```diff
@@ -78,4 +80,9 @@ namespace reinforcement_learning {
     *retval = new console_tracer();
     return error_code::success;
   }
+
+  int binding_tracer_create(i_trace** retval, const u::configuration& cfg, i_trace* trace_logger, api_status* status) {
+    *retval = trace_logger;
```

Comment:
>*retval = trace_logger; [](start = 4, length = 23)

What is the point of this as nothing is being created?

Reply:
Calling this method was just to conform the factory pattern. If you look at the usage of other logger, null_trace_create and console_trace_create, i_trace parameter is not used. Both of existing methods ingore it and returns a new instance.

Also the design of this create method to take i_trace as parameter and returns i_trace, to me, is a bit questionable.


---
In reply to: [252721602](https://github.com/VowpalWabbit/reinforcement_learning/pull/38#discussion_r252721602) [](ancestors = 252721602)

Reply:
To NY team, check the comments in live_model.cc. Let's have a single place to discuss the factory interfaces.

---
In reply to: [252721602](https://github.com/VowpalWabbit/reinforcement_learning/pull/38#discussion_r252721602) [](ancestors = 252721602)

Reply:
A couple of reasons for passing in i_trace to trace logger factory.  1) creation of user trace logging facility can fail.  We would like to log this somehow.  2) possible future chained logging support 

---
In reply to: [252881488](https://github.com/VowpalWabbit/reinforcement_learning/pull/38#discussion_r252881488) [](ancestors = 252881488,252721602)